src/UIHandler.h

```diff
@@ -0,0 +1,110 @@
+#pragma once
+#ifndef UIHANDLER_H
+#define UIHANDLER_H
+
+#include "debug.h"
+
+#include "cursesdef.h"
+#include "output.h"
+
+struct IntPair
+{
+    int x;
+    int y;
+};
+
+class UIPanel
+{
+public:
+    virtual std::vector<UIPanel*> getChild() = 0;
+    virtual void addChild(UIPanel *panel) = 0;
+    virtual void removeChild(size_t index) = 0; // Passes back removed child
+
+    virtual IntPair RequestedSize(bool min) = 0;
+    virtual int SetSize(IntPair size) = 0;
+};
+/*
+// A generic panel to split crap
+class UISplitPanel
+{
+public:
+    enum class Arangments
+    {
+        Stacked,
+        SideBySide,
+
+        Undefined = 0,
+        Total
+    };
+
+    UISplitPanel(Arangments arangment);
+
+    std::vector<UIPanel*> getChild();
+    void addChild(UIPanel *panel);
+
+    // Passes back removed child
+    // We swap this with the last then push back
+    // Previous Indexes are not maintained!!!
+    void removeChild(size_t index);
+
+    IntPair RequestedSize(bool min);
+    int SetSize(IntPair size);
+private:
+    std::vector<UIPanel*> m_childPanels;
+    Arangments m_arangment;
+};
+*/
+// to be used _ONLY_ in windows
+// If you want a generic panel use PaddingPanel
+class UIParentPanel : public UIPanel
+{
+public:
+    std::vector<UIPanel*> getChild();
+    void addChild(UIPanel *panel);
+
+    // Passes back removed child
+    // We swap this with the last then push back
+    // Previous Indexes are not maintained!!!
+    void removeChild(size_t index);
+
+    IntPair RequestedSize(bool min);
+    int SetSize(IntPair size);
+private:
+    IntPair m_thisSize;
+    //IntPair m_lastSize;
+
+    std::vector<UIPanel*> m_childPanels;
+};
+
+class UIWindow {
+public:
+    // TODO REMOVE EVENTUALY!!!
+    WINDOW* LegacyWindow() {return m_wf_win;};
+
+
+    enum class Location
+    {
+        Centered,
+
+        Undefined = 0,
```

Comment:
Is it on purpose that `Undefined` and `Centered` have the same value? If so, you might want to make this explicit: `Undefined = Centered,`.